### PR TITLE
Updated Integrals

### PR DESCRIPTION
**Pull request by ndmeier25**
Reigned in my tendency to go overboard and only merged pre-existing files, with the exception of the double precision sphericald and pVpd files (taken directly), sortUtil.h, and several Becke grid functions, resulting in a working version with the new integral code! No DFT, SCF, CAS, x2c, or other special features at the moment.

### CODE REVIEW

File: examples/geom_timings/main.cpp [hunk 1]
```diff
@@ -72,42 +72,42 @@ int main(int argc, char* argv[]) {
...
+    double * pVp = new double[N2];
     int prl = 0;
     #pragma acc enter data \
       create(A[0:Naux2],C[0:N2a],S[0:N2],En[0:N2],T[0:N2],pVp[0:N2])
     printf("1e ints: %d\n2c2e ints: %d\n3c3e ints: %d\n",N2, Naux2, N2a);
     auto t1 = chrono::high_resolution_clock::now();
     compute_ST(natoms, atno, coordsf, basis, nrad, size_ang, ang_g, ang_w, S, T, prl);
     auto t2 = chrono::high_resolution_clock::now();
-    compute_all_2c_v2(natoms,atno,coordsf,basis_aux,nrad,size_ang,ang_g,ang_w,A,prl);
+    compute_all_2c_v2(0,natoms,atno,coordsf,basis_aux,nrad,size_ang,ang_g,ang_w,A,prl);
```
Comment (alstark1) [inline examples/geom_timings/main.cpp:95]: What purpose does this change serve? also with the compute_all_3c_para function

File: include/braggslater.h [hunk 2]
```diff
@@ -1,71 +1,86 @@
...
   return val;
 }
 
 #if USE_ACC
 //#pragma acc routine seq
 #endif
-FP1 get_radii_2(int Z)
+float get_radii_2(int Z)
 {
+  if (Z<=0) Z = 1;
```
Comment (alstark1) [inline include/braggslater.h:53]: Looks like this is not a correction, but new code.

File: include/braggslater.h [hunk 3]
```diff
@@ -1,71 +1,86 @@
...
   else if (Z<=10) val1 = bsf(Z-3,2,0);
   else if (Z<=18) val1 = bsf(Z-11,8,2);
   else if (Z<=20) val1 = bsf(Z-19,8,10);
   else if (Z<=30) val1 = bsf(1,Z-12,10);
   else if (Z<=36) val1 = bsf(Z-29,18,10);
 
-  FP1 r1 = val0*val0/(Z-val1);
+  float r1 = val0*val0/(Z-val1);
+
+  if (Z==3)
```
Comment (alstark1) [inline include/braggslater.h:78]: same thing here, new code

File: include/integrals.h
Comment (alstark1) [inline include/integrals.h:1]: I trust this, we should talk about this file specifically so I can get more info.

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -1,6 +1,9 @@
 add_subdirectory(integrals)
 add_subdirectory(libio)
 
-if(DO_GTO)
+if(${DO_GTO})
```
Comment (alstark1) [inline src/CMakeLists.txt:4]: curious about syntax here

File: src/integrals/CMakeLists.txt [hunk 5]
```diff
@@ -1,22 +1,30 @@
...
 
 add_library(SlaterGPU STATIC "${INTEGRALS_SOURCES}")
 target_include_directories(SlaterGPU PUBLIC
     "${CMAKE_SOURCE_DIR}/include/SlaterGPU"
+    "${CMAKE_SOURCE_DIR}/include"
+    "${CMAKE_SOURCE_DIR}/src/libcintw/"
+    "${CMAKE_SOURCE_DIR}/src/libio/"
     )
 
+add_definitions("-w")
```
Comment (alstark1) [inline src/integrals/CMakeLists.txt:26]: what does -w do here?

File: src/integrals/Vinr.cpp [hunk 6]
```diff
@@ -1,64 +1,112 @@
 #include "Vinr.h"
-
-// for getting cutoff
-// #include "get_rcutoff.h"
-// sorting and search
-#if TEST_SORT
-#include "lib/thrust_util.h"
```
Comment (alstark1) [inline src/integrals/Vinr.cpp:7]: what did this thrust_utils.h do?

File: src/integrals/integrals.cpp
Comment (alstark1) [inline src/integrals/integrals.cpp:1]: Need to talk about this file on its own

File: src/integrals/integrals_d.cpp [hunk 7]
```diff
@@ -1,41 +1,16 @@
...
 #include <string>
-// void auto_crash();
-// void print_duration(chrono::high_resolution_clock::time_point t1, chrono::high_resolution_clock::time_point t2, string name);
+void auto_crash();
+void print_duration(chrono::high_resolution_clock::time_point t1, chrono::high_resolution_clock::time_point t2, string name);
+void copy_to_all_gpu(int ngpu, int s1, double* A, int include_first);
 
-#if USE_ACC
-void copy_to_all_gpu(int ngpu, int s1, FP2* A, int include_first)
-{
```
Comment (alstark1) [inline src/integrals/integrals_d.cpp:15]: why get rid of this chunk?

File: src/integrals/integrals_d.cpp
Comment (alstark1) [inline src/integrals/integrals_d.cpp:1]: was compute_all_3c_para moved elsewhere?

File: src/integrals/pVp.cpp [hunk 8]
```diff
@@ -2735,65 +2772,286 @@ void get_dp_gp3(int gs, FP1* grid, FP1* val)
...
-    val[3*i+1] *= 5.f*(x4*x2-10.f*x4*y2+5.f*x2*y4+(x4-6.f*x2*y2+y4)*z2)*rm7;
-    val[3*i+2] *= -5.f*y*z*(5.f*x4-10.f*x2*y2+y4)*rm7;
+    val[3*i+0] *= 9.f*x4*y2+6.f*x2*y4-3.f*y4*y2+21*(x4-6.f*x2*y2+y4)*z2+24.f*(-x2+y2)*z4*rm7;
+    val[3*i+1] *= 3.f*x*y*(-3.f*x4-2*x2*y2+y4+28.f*(x-y)*(x+y)*z2+16.f*z4)*rm7;
+    val[3*i+2] *= -3.f*x*(x2-3.f*y2)*z*(7.f*(x2+y2)-8.f*z2)*rm7;
   }
   return;
 }
 
-void eval_dp_3r(int gs, FP1* grid, FP1* val, int n1, int l1, int m1)
```
Comment (alstark1) [inline src/integrals/pVp.cpp:2786]: what are these function definitions that had nothing defined?

File: src/libio/read.cpp
Comment (alstark1) [inline src/libio/read.cpp:1]: should tlak about functions added here.

File: src/libio/write.cpp [hunk 9]
```diff
@@ -197,19 +194,28 @@ void get_nxyzr(int n1, int l1, int m1, int& nx, int& ny, int& nz, int& nr)
       if (m1== 2) { nx = 2; ny = 2; }
      #endif
     }
+    else if (l1==3)
```
Comment (alstark1) [inline src/libio/write.cpp:197]: what is this change?

**Review by alstark1 — COMMENTED** [review]
Looks good, curious to go over some parts of the code though in person.

File: src/libio/write.cpp [hunk 10]
```diff
@@ -132,9 +129,9 @@ void get_nxyzr(int n1, int l1, int m1, int& nx, int& ny, int& nz, int& nr)
       nr = 1;
     else if (l1==1)
     {
-      if (m1==0)      nx = 1;
-      else if (m1==1) ny = 1;
-      else            nz = 1;
+      if (m1==0)      nz = 1;
+      else if (m1==1) nx = 1;
+      else            ny = 1;
```
[the remark this answers is not in the document]
Comment (joshkamm) [reply src/libio/write.cpp:134]: Just talked to Nate in person. It seems like the answer is yes.

**Pull request merged**